src/environmentd/src/environmentd/main.rs

```diff
@@ -1026,8 +1053,18 @@ fn run(mut args: Args) -> Result<(), anyhow::Error> {
         },
     };
 
-    let cluster_replica_sizes: ClusterReplicaSizeMap =
+    let mut cluster_replica_sizes: ClusterReplicaSizeMap =
         serde_json::from_str(&args.cluster_replica_sizes).context("parsing replica size map")?;
+    if !license_key.allow_credit_consumption_override {
+        for (name, replica) in cluster_replica_sizes.0.iter_mut() {
+            let Some(memory_limit) = replica.memory_limit else {
+                panic!("No memory limit found in cluster definition for {name}");
+            };
+            replica.credits_per_hour = Numeric::from(
+                (memory_limit.0 * replica.scale * u64::try_from(replica.workers).unwrap()).0,
+            ) / Numeric::from(1024 * 1024 * 1024);
```

Comment:
Mind documenting the use of `1024 * 1024 * 1024`? It seems to assume that memory_limit values are in GiB? Also FWIW we generally try to use the `checked_mul(...)` or `saturating_mul(...)` methods instead of just `*`